From: Stained Glass OS <ke7oxh@gmail.com>
Subject: [PATCH] msvcp: Export the members whose names differ only in access.

Exports whose names differ from implemented ones only in member access
(private, protected or public): msvcp70's private basic_string helpers
_Copy, _Eos, _Grow, _Inside and _Myptr (char, wchar_t, unsigned short),
msvcp60's time_get destructors, and msvcp70/msvcp71's public
ios_base::_Addstd. Also maps ARM names whose i386 twins are implemented.
Each name has the same function, argument list and calling convention
as the one it matches.
---
 dlls/msvcp60/msvcp60.spec |  8 ++---
 dlls/msvcp70/msvcp70.spec | 76 +++++++++++++++++++++++------------------------
 dlls/msvcp71/msvcp71.spec |  6 ++--
 3 files changed, 45 insertions(+), 45 deletions(-)

diff --git a/dlls/msvcp60/msvcp60.spec b/dlls/msvcp60/msvcp60.spec
index fbf281e..4588047 100644
--- a/dlls/msvcp60/msvcp60.spec
+++ b/dlls/msvcp60/msvcp60.spec
@@ -1290,11 +1290,11 @@
 @ thiscall -arch=i386 ??1?$numpunct@G@std@@UAE@XZ(ptr) numpunct_wchar_dtor
 @ cdecl -arch=win64 ??1?$numpunct@G@std@@UEAA@XZ(ptr) numpunct_wchar_dtor
 @ stub -arch=arm ??1?$time_get@DV?$istreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@UAA@XZ
-@ stub -arch=i386 ??1?$time_get@DV?$istreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@UAE@XZ
-@ stub -arch=win64 ??1?$time_get@DV?$istreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@UEAA@XZ
+@ thiscall -arch=i386 ??1?$time_get@DV?$istreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@UAE@XZ(ptr) time_get_char_dtor
+@ cdecl -arch=win64 ??1?$time_get@DV?$istreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@UEAA@XZ(ptr) time_get_char_dtor
 @ stub -arch=arm ??1?$time_get@GV?$istreambuf_iterator@GU?$char_traits@G@std@@@std@@@std@@UAA@XZ
-@ stub -arch=i386 ??1?$time_get@GV?$istreambuf_iterator@GU?$char_traits@G@std@@@std@@@std@@UAE@XZ
-@ stub -arch=win64 ??1?$time_get@GV?$istreambuf_iterator@GU?$char_traits@G@std@@@std@@@std@@UEAA@XZ
+@ thiscall -arch=i386 ??1?$time_get@GV?$istreambuf_iterator@GU?$char_traits@G@std@@@std@@@std@@UAE@XZ(ptr) time_get_wchar_dtor
+@ cdecl -arch=win64 ??1?$time_get@GV?$istreambuf_iterator@GU?$char_traits@G@std@@@std@@@std@@UEAA@XZ(ptr) time_get_wchar_dtor
 @ cdecl -arch=arm ??1?$time_put@DV?$ostreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@UAA@XZ(ptr) time_put_char_dtor
 @ thiscall -arch=i386 ??1?$time_put@DV?$ostreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@UAE@XZ(ptr) time_put_char_dtor
 @ cdecl -arch=win64 ??1?$time_put@DV?$ostreambuf_iterator@DU?$char_traits@D@std@@@std@@@std@@UEAA@XZ(ptr) time_put_char_dtor
diff --git a/dlls/msvcp70/msvcp70.spec b/dlls/msvcp70/msvcp70.spec
index 53a211f..f626061 100644
--- a/dlls/msvcp70/msvcp70.spec
+++ b/dlls/msvcp70/msvcp70.spec
@@ -2377,9 +2377,9 @@
 @ cdecl -arch=arm ?_Addfac@locale@std@@QAAAAV12@PAVfacet@12@II@Z(ptr ptr long long) locale__Addfac
 @ thiscall -arch=i386 ?_Addfac@locale@std@@QAEAAV12@PAVfacet@12@II@Z(ptr ptr long long) locale__Addfac
 @ cdecl -arch=win64 ?_Addfac@locale@std@@QEAAAEAV12@PEAVfacet@12@_K1@Z(ptr ptr long long) locale__Addfac
-@ stub -arch=arm ?_Addstd@ios_base@std@@QAAXXZ
-@ stub -arch=i386 ?_Addstd@ios_base@std@@QAEXXZ
-@ stub -arch=win64 ?_Addstd@ios_base@std@@QEAAXXZ
+@ cdecl -arch=arm ?_Addstd@ios_base@std@@QAAXXZ(ptr) ios_base_Addstd
+@ thiscall -arch=i386 ?_Addstd@ios_base@std@@QAEXXZ(ptr) ios_base_Addstd
+@ cdecl -arch=win64 ?_Addstd@ios_base@std@@QEAAXXZ(ptr) ios_base_Addstd
 @ extern -arch=win32 ?_BADOFF@std@@3JB std_BADOFF
 @ extern -arch=win64 ?_BADOFF@std@@3_JB std_BADOFF
 @ cdecl -arch=arm ?_Callfns@ios_base@std@@AAAXW4event@12@@Z(ptr long) ios_base_Callfns
@@ -2393,12 +2393,12 @@
 # extern -arch=win32 ?_Cm@?5??log@std@@YA?AV?$complex@M@2@ABV32@@Z@4MB
 # extern -arch=win32 ?_Cm@?5??log@std@@YA?AV?$complex@N@2@ABV32@@Z@4NB
 # extern -arch=win32 ?_Cm@?5??log@std@@YA?AV?$complex@O@2@ABV32@@Z@4OB
-@ stub -arch=arm ?_Copy@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAAXII@Z
-@ stub -arch=i386 ?_Copy@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAEXII@Z
-@ stub -arch=arm ?_Copy@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAAXII@Z
-@ stub -arch=i386 ?_Copy@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAEXII@Z
-@ stub -arch=arm ?_Copy@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAAXII@Z
-@ stub -arch=i386 ?_Copy@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAEXII@Z
+@ cdecl -arch=arm ?_Copy@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAAXII@Z(ptr long long) basic_string_char__Copy
+@ thiscall -arch=i386 ?_Copy@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAEXII@Z(ptr long long) basic_string_char__Copy
+@ cdecl -arch=arm ?_Copy@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAAXII@Z(ptr long long) basic_string_wchar__Copy
+@ thiscall -arch=i386 ?_Copy@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAEXII@Z(ptr long long) basic_string_wchar__Copy
+@ cdecl -arch=arm ?_Copy@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAAXII@Z(ptr long long) basic_string_wchar__Copy
+@ thiscall -arch=i386 ?_Copy@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAEXII@Z(ptr long long) basic_string_wchar__Copy
 @ cdecl ?_Cosh@?$_Ctraits@M@std@@SAMMM@Z(float float) _FCosh
 @ cdecl ?_Cosh@?$_Ctraits@N@std@@SANNN@Z(double double) _Cosh
 @ cdecl ?_Cosh@?$_Ctraits@O@std@@SAOOO@Z(double double) _Cosh
@@ -2414,14 +2414,14 @@
 @ cdecl -arch=arm ?_Endwrite@?$basic_filebuf@_WU?$char_traits@_W@std@@@std@@IAA_NXZ(ptr) basic_filebuf_wchar__Endwrite
 @ thiscall -arch=i386 ?_Endwrite@?$basic_filebuf@_WU?$char_traits@_W@std@@@std@@IAE_NXZ(ptr) basic_filebuf_wchar__Endwrite
 @ cdecl -arch=win64 ?_Endwrite@?$basic_filebuf@_WU?$char_traits@_W@std@@@std@@IEAA_NXZ(ptr) basic_filebuf_wchar__Endwrite
-@ stub -arch=arm ?_Eos@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAAXI@Z
-@ stub -arch=i386 ?_Eos@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAEXI@Z
-@ stub -arch=win64 ?_Eos@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEAAX_K@Z
-@ stub -arch=arm ?_Eos@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAAXI@Z
-@ stub -arch=i386 ?_Eos@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAEXI@Z
-@ stub -arch=win64 ?_Eos@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AEAAX_K@Z
-@ stub -arch=arm ?_Eos@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAAXI@Z
-@ stub -arch=i386 ?_Eos@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAEXI@Z
+@ cdecl -arch=arm ?_Eos@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAAXI@Z(ptr long) basic_string_char_eos
+@ thiscall -arch=i386 ?_Eos@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAEXI@Z(ptr long) basic_string_char_eos
+@ cdecl -arch=win64 ?_Eos@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEAAX_K@Z(ptr long) basic_string_char_eos
+@ cdecl -arch=arm ?_Eos@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAAXI@Z(ptr long) basic_string_wchar_eos
+@ thiscall -arch=i386 ?_Eos@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAEXI@Z(ptr long) basic_string_wchar_eos
+@ cdecl -arch=win64 ?_Eos@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AEAAX_K@Z(ptr long) basic_string_wchar_eos
+@ cdecl -arch=arm ?_Eos@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAAXI@Z(ptr long) basic_string_wchar_eos
+@ thiscall -arch=i386 ?_Eos@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAEXI@Z(ptr long) basic_string_wchar_eos
 @ cdecl -arch=win32 ?_Exp@?$_Ctraits@M@std@@SAFPAMMF@Z(ptr float long) _FExp
 @ cdecl -arch=win64 ?_Exp@?$_Ctraits@M@std@@SAFPEAMMF@Z(ptr float long) _FExp
 @ cdecl -arch=win32 ?_Exp@?$_Ctraits@N@std@@SAFPANNF@Z(ptr double long) _Exp
@@ -2620,9 +2620,9 @@
 @ cdecl -arch=arm ?_Grow@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAA_NI_N@Z(ptr long long) basic_string_char_grow
 @ thiscall -arch=i386 ?_Grow@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAE_NI_N@Z(ptr long long) basic_string_char_grow
 @ cdecl -arch=win64 ?_Grow@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AEAA_N_K_N@Z(ptr long long) basic_string_char_grow
-@ stub -arch=arm ?_Grow@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAA_NI_N@Z
-@ stub -arch=i386 ?_Grow@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAE_NI_N@Z
-@ stub -arch=win64 ?_Grow@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AEAA_N_K_N@Z
+@ cdecl -arch=arm ?_Grow@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAA_NI_N@Z(ptr long long) basic_string_wchar_grow
+@ thiscall -arch=i386 ?_Grow@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAE_NI_N@Z(ptr long long) basic_string_wchar_grow
+@ cdecl -arch=win64 ?_Grow@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AEAA_N_K_N@Z(ptr long long) basic_string_wchar_grow
 @ cdecl -arch=arm ?_Grow@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAA_NI_N@Z(ptr long long) basic_string_wchar_grow
 @ thiscall -arch=i386 ?_Grow@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAE_NI_N@Z(ptr long long) basic_string_wchar_grow
 @ extern ?_Id_cnt@id@locale@std@@0HA locale_id__Id_cnt
@@ -2800,12 +2800,12 @@
 @ cdecl -arch=arm ?_Initcvt@?$basic_filebuf@_WU?$char_traits@_W@std@@@std@@IAAXPAV?$codecvt@_WDH@2@@Z(ptr ptr) basic_filebuf_wchar__Initcvt_cvt
 @ thiscall -arch=i386 ?_Initcvt@?$basic_filebuf@_WU?$char_traits@_W@std@@@std@@IAEXPAV?$codecvt@_WDH@2@@Z(ptr ptr) basic_filebuf_wchar__Initcvt_cvt
 @ cdecl -arch=win64 ?_Initcvt@?$basic_filebuf@_WU?$char_traits@_W@std@@@std@@IEAAXPEAV?$codecvt@_WDH@2@@Z(ptr ptr) basic_filebuf_wchar__Initcvt_cvt
-@ stub -arch=arm ?_Inside@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAA_NPBD@Z
-@ stub -arch=i386 ?_Inside@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAE_NPBD@Z
-@ stub -arch=arm ?_Inside@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAA_NPBG@Z
-@ stub -arch=i386 ?_Inside@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAE_NPBG@Z
-@ stub -arch=arm ?_Inside@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAA_NPB_W@Z
-@ stub -arch=i386 ?_Inside@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAE_NPB_W@Z
+@ cdecl -arch=arm ?_Inside@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAA_NPBD@Z(ptr str) basic_string_char_inside
+@ thiscall -arch=i386 ?_Inside@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAE_NPBD@Z(ptr str) basic_string_char_inside
+@ cdecl -arch=arm ?_Inside@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAA_NPBG@Z(ptr ptr) basic_string_wchar_inside
+@ thiscall -arch=i386 ?_Inside@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAE_NPBG@Z(ptr ptr) basic_string_wchar_inside
+@ cdecl -arch=arm ?_Inside@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAA_NPB_W@Z(ptr ptr) basic_string_wchar_inside
+@ thiscall -arch=i386 ?_Inside@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAE_NPB_W@Z(ptr ptr) basic_string_wchar_inside
 @ cdecl -arch=arm ?_Ipfx@?$basic_istream@DU?$char_traits@D@std@@@std@@QAA_N_N@Z(ptr long) basic_istream_char__Ipfx
 @ thiscall -arch=i386 ?_Ipfx@?$basic_istream@DU?$char_traits@D@std@@@std@@QAE_N_N@Z(ptr long) basic_istream_char__Ipfx
 @ cdecl -arch=win64 ?_Ipfx@?$basic_istream@DU?$char_traits@D@std@@@std@@QEAA_N_N@Z(ptr long) basic_istream_char__Ipfx
@@ -2847,18 +2847,18 @@
 @ cdecl -arch=arm ?_Makpat@?$_Mpunct@_W@std@@AAAXAAUpattern@money_base@2@III@Z(ptr ptr long long long) mpunct_wchar__Makpat
 @ thiscall -arch=i386 ?_Makpat@?$_Mpunct@_W@std@@AAEXAAUpattern@money_base@2@III@Z(ptr ptr long long long) mpunct_wchar__Makpat
 @ cdecl -arch=win64 ?_Makpat@?$_Mpunct@_W@std@@AEAAXAEAUpattern@money_base@2@III@Z(ptr ptr long long long) mpunct_wchar__Makpat
-@ stub -arch=arm ?_Myptr@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAAPADXZ
-@ stub -arch=i386 ?_Myptr@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAEPADXZ
-@ stub -arch=arm ?_Myptr@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@ABAPBDXZ
-@ stub -arch=i386 ?_Myptr@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@ABEPBDXZ
-@ stub -arch=arm ?_Myptr@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAAPAGXZ
-@ stub -arch=i386 ?_Myptr@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAEPAGXZ
-@ stub -arch=arm ?_Myptr@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@ABAPBGXZ
-@ stub -arch=i386 ?_Myptr@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@ABEPBGXZ
-@ stub -arch=arm ?_Myptr@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAAPA_WXZ
-@ stub -arch=i386 ?_Myptr@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAEPA_WXZ
-@ stub -arch=arm ?_Myptr@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@ABAPB_WXZ
-@ stub -arch=i386 ?_Myptr@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@ABEPB_WXZ
+@ cdecl -arch=arm ?_Myptr@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAAPADXZ(ptr) basic_string_char_ptr
+@ thiscall -arch=i386 ?_Myptr@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@AAEPADXZ(ptr) basic_string_char_ptr
+@ cdecl -arch=arm ?_Myptr@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@ABAPBDXZ(ptr) basic_string_char_const_ptr
+@ thiscall -arch=i386 ?_Myptr@?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@ABEPBDXZ(ptr) basic_string_char_const_ptr
+@ cdecl -arch=arm ?_Myptr@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAAPAGXZ(ptr) basic_string_wchar_ptr
+@ thiscall -arch=i386 ?_Myptr@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@AAEPAGXZ(ptr) basic_string_wchar_ptr
+@ cdecl -arch=arm ?_Myptr@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@ABAPBGXZ(ptr) basic_string_wchar_const_ptr
+@ thiscall -arch=i386 ?_Myptr@?$basic_string@GU?$char_traits@G@std@@V?$allocator@G@2@@std@@ABEPBGXZ(ptr) basic_string_wchar_const_ptr
+@ cdecl -arch=arm ?_Myptr@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAAPA_WXZ(ptr) basic_string_wchar_ptr
+@ thiscall -arch=i386 ?_Myptr@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@AAEPA_WXZ(ptr) basic_string_wchar_ptr
+@ cdecl -arch=arm ?_Myptr@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@ABAPB_WXZ(ptr) basic_string_wchar_const_ptr
+@ thiscall -arch=i386 ?_Myptr@?$basic_string@_WU?$char_traits@_W@std@@V?$allocator@_W@2@@std@@ABEPB_WXZ(ptr) basic_string_wchar_const_ptr
 @ cdecl ?_Nanv@?$_Ctraits@M@std@@SAMM@Z(float) std_Ctraits_float__Nanv
 @ cdecl ?_Nanv@?$_Ctraits@N@std@@SANN@Z(double) std_Ctraits_double__Nanv
 @ cdecl ?_Nanv@?$_Ctraits@O@std@@SAOO@Z(double) std_Ctraits_double__Nanv
diff --git a/dlls/msvcp71/msvcp71.spec b/dlls/msvcp71/msvcp71.spec
index 7adb820..3ed2ca2 100644
--- a/dlls/msvcp71/msvcp71.spec
+++ b/dlls/msvcp71/msvcp71.spec
@@ -2602,9 +2602,9 @@
 @ cdecl -arch=arm ?_Addfac@locale@std@@QAAAAV12@PAVfacet@12@II@Z(ptr ptr long long) locale__Addfac
 @ thiscall -arch=i386 ?_Addfac@locale@std@@QAEAAV12@PAVfacet@12@II@Z(ptr ptr long long) locale__Addfac
 @ cdecl -arch=win64 ?_Addfac@locale@std@@QEAAAEAV12@PEAVfacet@12@_K1@Z(ptr ptr long long) locale__Addfac
-@ stub -arch=arm ?_Addstd@ios_base@std@@QAAXXZ
-@ stub -arch=i386 ?_Addstd@ios_base@std@@QAEXXZ
-@ stub -arch=win64 ?_Addstd@ios_base@std@@QEAAXXZ
+@ cdecl -arch=arm ?_Addstd@ios_base@std@@QAAXXZ(ptr) ios_base_Addstd
+@ thiscall -arch=i386 ?_Addstd@ios_base@std@@QAEXXZ(ptr) ios_base_Addstd
+@ cdecl -arch=win64 ?_Addstd@ios_base@std@@QEAAXXZ(ptr) ios_base_Addstd
 @ extern -arch=win32 ?_BADOFF@std@@3JB std_BADOFF
 @ extern -arch=win64 ?_BADOFF@std@@3_JB std_BADOFF
 @ cdecl -arch=arm ?_Callfns@ios_base@std@@AAAXW4event@12@@Z(ptr long) ios_base_Callfns
